skimage/restoration/_denoise.py

```diff
@@ -594,26 +600,39 @@ def denoise_wavelet(image, sigma=None, wavelet='db1', mode='soft',
             ('Invalid method: {}. The currently supported methods are '
              '"BayesShrink" and "VisuShrink"').format(method))
 
-    image = img_as_float(image)
-
     if multichannel:
         if isinstance(sigma, numbers.Number) or sigma is None:
             sigma = [sigma] * image.shape[-1]
 
+    clip_output = False
+    if image.dtype.kind != 'f':
+        clip_output = True
+        data_range_pre = image.max() - image.min()
+        image = img_as_float(image)
+        data_range_post = image.max() - image.min()
+        scale_factor = data_range_post / data_range_pre
+        if sigma is not None:
+            if multichannel:
+                sigma = [s * scale_factor for s in sigma]
+            else:
+                sigma *= scale_factor
```

Comment:
I agree that it may warrant that. It is clearly a change in behavior, but I think will be more intuitive going forward.

`estimate_sigma` does not do any rescaling, so an input in range [0, 1e6] might produce a huge sigma value of 1e4 or so which would clearly give a crazy oversmoothing when applied to an image that has been rescaled to range [0, 1]!

For the current release, `sigma` only really works if the user computed it for an `image` scaled to [0, 1] or [1, -1] (or if it is left to the default of `None`).

Reply:
Oh yes, I am absolutely not arguing that we shouldn't have the change, just whether it should be a silent/hard change or a deprecation. I think we might have many people who had their sigma set (empirically) 255 times lower than intended, and now they will find that their code does no denoising at all, and be at a loss as to why.

Reply:
I think the `clip_output` flag to avoid clipping floating point outputs to [0, 1] or [-1, 1] should be applied now as a bug fix.

I think the automatic rescaling of sigma when the image must be rescaled is convenient and the right thing to do, but that could potentially go through a deprecation cycle for the reason you mention.

Reply:
Frustrating—so close to a bug fix, but not quite!  We may argue for an accelerated deprecation in this case, perhaps; i.e., that the behavior changes immediately but that old users at least get a warning?  Either that, or the more conservative option of deprecating.

Reply:
@stefanv I'd be on board with "accelerated deprecation" in cases of bug fixes that can affect correct running of code. @grlee77 what do you think? My main concern with this kind of warning is that it should be easy to disable.

Reply:
I would lean toward "accelerated deprecation".  I think it is likely not uncommon that users with floating point data would be getting bad results currently.

> My main concern with this kind of warning is that it should be easy to disable.

Is there a recommended way to do this? I am only familiar with the user needing to use `with warnings.catch_warnings` context to disable the unwanted warnings?

Reply:
I'm thinking some kind of long-lived keyword argument, `rescale_sigma=None`, which means `True`, and if set to None (default), a warning is issued, but if set to `True` there is no warning. If set to `False` you get the old behaviour but a warning that this option will go away in the future. In both the docstring and the warning, explain the issue and maybe link to this discussion.

Thoughts?

The point is that `catch_warnings` is super-verbose and people are unlikely to want to pollute their codebase with it.

The main problem with the long-lived kwarg is that it results in yet another deprecation in the future. I don't see a better way around it though...

Reply:
@JDWarner tends to have opinions about these things... =)

Reply:
I think adding a `rescale_sigma=None`argument sounds good and gives an easier way to bypass any warning if desired.